pulp_rpm/app/migrations/0049_profiles_fix.py

```diff
@@ -0,0 +1,39 @@
+from django.db import migrations
+import yaml
+
+
+def fixup_modulemd_profiles(apps, schema_editor):
+    """ Set "profiles" for any saved post-3.19 back to the 3.18 format.
+    """
+    Modulemd = apps.get_model("rpm", "Modulemd")
+
+    modules_to_update = []
+
+    for mmd in Modulemd.objects.all().iterator():
+        modulemd = yaml.safe_load(mmd.snippet)
+        unprocessed_profiles = modulemd["data"].get("profiles", {})
+        profiles = {}
+        if unprocessed_profiles:
+            for name, data in unprocessed_profiles.items():
+                rpms = data.get("rpms")
+                if rpms:
+                    profiles[name] = rpms
+            mmd.profiles = profiles
```

Comment:
should this be outside of if block?

Reply:
I read this as "we only set profiles if there any unprocessed (ie old-style) ones found" - is that correct @dralley ?

Reply:
in that case there is no need to append such modules for update( line 23)  since nothing will be changed in the profiles

Reply:
@ipanova OK, I see your point there. So if we moved line#21 to inside the if: at line#16, we'd only add 'mmd' to 'modules_to_update' if that 'mmd' has unprocessed-modules. And then only mmd's whose profiles had been modified, would be considered in the bulk_update() calls. Does tha make sense?

Reply:
I think my overall thought was that they pretty much all have profiles so it wouldn't make much difference, butI have no objection to making it conditional.